### PR TITLE
Piece movement

### PR DESCRIPTION
**Pull request by EXC3ll3NTrhyTHM**
Updated a lot. It works and we reduced code smell.

We didnt really modify a whole lot of the existing functionality. I would say the most modification we did to a methods functionality would be the validateCellSelection and performMove function. These now have switch statements which do different things depending on the game phase. But for the PIECE_PLACEMENT game phase the original functionality remains.

The main thing we did was added a Phase enum to the Player class which tracks the game phase for each player independently and had performMove now update the Phase for each player based on the state of the game at the end of a turn. We also added Cori's finding of valid adjacent cells in the CellPane class. This method helps in validationCellSelection for the PIECE_MOVEMENT Phase.

### CODE REVIEW

**Review by SangJinHyun — COMMENTED** [review]
public boolean is(CellState state) { return this.cellState == state; }

Can we perhaps rename this?

**Comment by atparris** [comment]
There was an issue with one of the TestPiecePlacement methods. I fixed this, and also created a test file for pieceMovement. PR could probably be replaced with testPieceMovement branch (or at least incorporate test cases).

**Comment by EXC3ll3NTrhyTHM** [comment]
I think 

> { return this.cellState == state; }

Yes we can actually remove this entirely and replace with Java equals() method.

**Comment by EXC3ll3NTrhyTHM** [comment]
> public boolean is(CellState state) { return this.cellState == state; }
> 
> Can we perhaps rename this?

Yes we can actually replace it with Java equals() method. I believe

**Comment by EXC3ll3NTrhyTHM** [comment]
Better idea we overload java equals for the Cell class.

**Comment by EXC3ll3NTrhyTHM** [comment]
Even better idea we use the normal java equals without overloading and compare `cell.cellState` instead of `cell` to `getPlayerColorAsCellState`.

File: src/Morris_FX/Logic/Board.java [hunk 1]
```diff
@@ -1,137 +1,131 @@
...
+
+                invalidCellType = InvalidCellType.NONE;
+
+                return true;
+            case PIECE_MOVEMENT:
+                if (!currentPlayer.hasPieceToMove()) {
+                    if (cell.isEmpty()) {
+                        return false;
+                    }
+                    if (cell.canPickup(currentPlayer)) {
```
Comment (EXC3ll3NTrhyTHM) [inline src/Morris_FX/Logic/Board.java:65]: @Mrocor would we be able to change this to `currentPlayer.canPickup(cell)`?

**Review by ahrayahlabcoimin — APPROVED** [review]

**Review by atparris — APPROVED** [review]
Beautiful!

**Pull request merged**